in_toto/in_toto_sign.py

```
@@ -155,15 +162,13 @@ def _verify_metadata(metadata, args):
       pub_key_dict = gpg_interface.export_pubkeys(
           args.gpg, args.gpg_home)
 
-
-    for keyid, verification_key in pub_key_dict.items():
-      metadata.verify_signature(verification_key)
-      LOG.info("Signature verification passed for keyid '{}'"
-          .format(keyid))
+    keys = create_key_list_from_key_dict(pub_key_dict)
+    LOG.info("Verifying metadata signatures against keys...")
+    metadata.verify_signatures(keys, len(keys))
```

Prior to this PR, we failed if any of the public keys couldn't verify the signature. 

Now `create_key_list_from_key_dict` might silently drop keys that don't match the expected schemas and verification would still pass, as long as at least one key successfully verifies a signature.